waku/v2/protocol/filter/client.go

```diff
@@ -46,11 +47,25 @@ type WakuFilterLightNode struct {
 	pm            *peermanager.PeerManager
 }
 
-type WakuFilterPushResult struct {
+type WakuFilterPushError struct {
 	Err    error
 	PeerID peer.ID
 }
 
+type WakuFilterPushResult struct {
+	errs []WakuFilterPushError
+	mu   sync.Mutex
+}
```

Comment:
You could convert the WakuFilterPushResult to a mutex:
```suggestion
type WakuFilterPushResult struct {
        sync.RWMutex
	errs []WakuFilterPushError   
}
```